juju/controller.py

```diff
@@ -206,15 +206,21 @@ def kill(self):
         cloud = list(result.clouds.keys())[0]  # only lives on one cloud
         return tag.untag('cloud-', cloud)
 
-    def get_models(self, all_=False, username=None):
+    async def list_models(self, all_=False, username=None):
         """Return list of available models on this controller.
 
         :param bool all_: List all models, regardless of user accessibilty
             (admin use only)
         :param str username: User for which to list models (admin use only)
 
         """
-        pass
+        model_facade = client.ModelManagerFacade()
+        model_facade.connect(self.connection)
+        username = username or self.connection.info['user-info']['identity']
+
+        results = await model_facade.ListModels(tag.user(username))
+        models = results.user_models
+        return [(m.model.name, m.model.uuid) for m in models]
```

Comment:
I think that I have a slight preference for returning `[m.model for m in models]`, with a note that you can get .name, .uuid, and .owner from each model object. That way, if the websocket api ever decides to return more info about a model (e.g., we add some major model relevant feature), we get it for free.

Not a blocker, though, because I can see the argument against, as well as for, this comment :-)